### PR TITLE
OCPBUGS-27366: set the --cluster-profile flag for openshift/api rendering

### PR DESCRIPTION
**Pull request by deads2k**
requires https://github.com/openshift/api/pull/1734

### CODE REVIEW

File: data/data/bootstrap/files/usr/local/bin/bootkube.sh.template [hunk 1]
```diff
@@ -93,13 +93,16 @@ then
 
 	rm --recursive --force api-bootstrap
 
+  # we should distinguish between self-managed-high-availability and single-node-developer, but for now they are identical
```
Comment (deads2k) [inline data/data/bootstrap/files/usr/local/bin/bootkube.sh.template:96]: @patrickdillon any easy way to distinguish?

Comment (patrickdillon) [reply data/data/bootstrap/files/usr/local/bin/bootkube.sh.template:96]: Apparently there's a very easy way!
https://github.com/openshift/installer/blob/7be2cf71cdef9f5fe0d2d9687615c2bb2741ffcd/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template#L12-L16

Comment (deads2k) [reply data/data/bootstrap/files/usr/local/bin/bootkube.sh.template]: Awesome!  Thanks.

File: data/data/bootstrap/files/usr/local/bin/bootkube.sh.template [hunk 2]
```diff
@@ -93,13 +93,21 @@ then
 
 	rm --recursive --force api-bootstrap
 
+  # A more desireable end state would be require wiring up the infrastructure in some way and read it from rendered-manifest-dir
```
Comment (patrickdillon) [inline data/data/bootstrap/files/usr/local/bin/bootkube.sh.template:96]: This is available in [`InfrastructureStatus.InfrastructureTopology`](https://github.com/openshift/api/blob/59fa376de7cb668ddb95a7ee4e9879d7f6ca2767/config/v1/types_infrastructure.go#L107). Do you prefer the current implementation for now?

Comment (deads2k) [reply data/data/bootstrap/files/usr/local/bin/bootkube.sh.template:96]: > This is available in [`InfrastructureStatus.InfrastructureTopology`](https://github.com/openshift/api/blob/59fa376de7cb668ddb95a7ee4e9879d7f6ca2767/config/v1/types_infrastructure.go#L107). Do you prefer the current implementation for now?

Yeah, given scope of change, this is our best 4.15 solution.

**Pull request merged**